galasa-managers-parent/galasa-managers-cicsts-parent/dev.galasa.cicsts.manager.ivt/src/main/java/dev/galasa/cicsts/manager/ivt/CICSTSManagerIVT.java

```diff
@@ -0,0 +1,199 @@
+    * @throws NetworkException 
+    * @throws TerminalInterruptedException 
+    * @throws KeyboardLockedException 
+    * @throws TimeoutException 
+    */
+   @Test 
+   public void testGetApplid() throws TimeoutException, KeyboardLockedException, TerminalInterruptedException, NetworkException, FieldNotFoundException  {
+	   logger.info("Testing that the CICS TS Manager gets the correct APPLID for the CICS Region");
+	   String testApplid = cics.getApplid();
+	   logger.info("APPLID retrieved by the CICS TS Manager: " + testApplid);
+	   terminal.type("CEMT INQUIRE").enter().waitForKeyboard();
+	   String terminalScreen = terminal.retrieveScreen();
+	   assertThat(terminalScreen.contains(testApplid)).isTrue();
+   }
+   
+   /**
+    * Tests that the CICS TS Manager retrieves a CICS Resource
+    * @throws CicstsManagerException 
+    */
+   @Test
+   public void testCicsResource() throws CicstsManagerException  {
+	   logger.info("Testing that the CICS TS Manager retrieves a CICS Resource");
+	   ICicsResource cicsResource = cics.cicsResource();
+	   assertThat(cicsResource).isNotNull();
+   }
+   
+   /**
+    * Tests that the CICS TS Manager retrieves the correct job
+    * @throws ZosBatchException 
+    * @throws FieldNotFoundException 
+    * @throws NetworkException 
+    * @throws TerminalInterruptedException 
+    * @throws KeyboardLockedException 
+    * @throws TimeoutException 
+    */
+   @Test
+   public void testGetRegionJob() throws CicstsManagerException, ZosBatchException, TimeoutException, KeyboardLockedException, TerminalInterruptedException, NetworkException, FieldNotFoundException {
+	   logger.info("Testing that the CICS TS Manager finds the DSE CICS job for the CICS Region");
+	   logger.info("There should be an active job as the CICS Region is running");
+	   boolean exceptionThrown = false;
+	   IZosBatchJob regionJob = null;
+	   try {
+		   regionJob = cics.getRegionJob();
+	   } catch (CicstsManagerException e) {
+		   exceptionThrown = true;
+	   }
+	   assertThat(exceptionThrown).isFalse();
+	   assertThat(regionJob).isNotNull();	   
+   }
+   
+   /**
+    * Tests that the CICS Terminal in the CICS TS Manager retrieves the correct CICS Region
+    * @throws FieldNotFoundException 
+    * @throws NetworkException 
+    * @throws TerminalInterruptedException 
+    * @throws KeyboardLockedException 
+    * @throws TimeoutException 
+    */
+   @Test
+   public void testGetCicsRegion() throws TimeoutException, KeyboardLockedException, TerminalInterruptedException, NetworkException, FieldNotFoundException {
+	   logger.info("Testing that the CICS Terminal gets the correct CICS Region");
+	   String testCicsRegion = terminal.getCicsRegion().toString().replace("CICS Region[", "").replace("]", "");
+	   logger.info("CICS Region retrieved by the ICicsTerminal: " + testCicsRegion);
+	   terminal.type("CEMT INQUIRE").enter().waitForKeyboard();
+	   assertThat(terminal.retrieveScreen().contains(testCicsRegion)).isTrue();
```

Comment:
This is used at least twice in this class - could it be pulled out into a private method.  That way if the implementation of it ever changes we don't have to update multiple places

Reply:
I have put this into a private method at the bottom of the class.